### PR TITLE
[config] Support configuring additional_endpoints with env vars

### PR DESCRIPTION
**Pull request by olivielpeau**
### What does this PR do?

Adds support for configuring `additional_endpoints` & `logs_config.additional_endpoints` with env vars.

### Additional Notes

This only applies to the infra & log agent for now.

TODO: extend this to the other agents (apm & process)

### CODE REVIEW

File: releasenotes/notes/additional-endpoints-config-from-env-vars-a5db51083cbd310f.yaml
Comment (olivielpeau) [inline releasenotes/notes/additional-endpoints-config-from-env-vars-a5db51083cbd310f.yaml]: let's remove the release note: technically `additional endpoints` is a hidden feature that we don't want to advertise since it has billing implications. Users should only use it after they've contacted support.

**Review by olivielpeau — COMMENTED** [review]
Could you add tests on the logs config when additional endpoints are set in the config file directly? (I haven't seen any pre-existing ones)

(note: Github won't let me approve the PR since I opened it)

File: releasenotes/notes/additional-endpoints-config-from-env-vars-a5db51083cbd310f.yaml
Comment (prognant) [reply releasenotes/notes/additional-endpoints-config-from-env-vars-a5db51083cbd310f.yaml]: It makes sense, release note dropped.

**Comment by prognant** [comment]
Good point as #4843 (https://github.com/DataDog/datadog-agent/pull/4843/files#diff-091ace02b49a46f55f2586cfec89a160R855)  broke the logs_config.additional_endpoints parsing as it does not follow the same structure as other *.additional_endpoints.
> Could you add tests on the logs config when additional endpoints are set in the config file directly? (I haven't seen any pre-existing ones)
> 
> (note: Github won't let me approve the PR since I opened it)

File: pkg/logs/config/config.go
Comment (hush-hush) [inline pkg/logs/config/config.go]: Could we make this a helper function and avoid code duplication from line 165-171 ?

File: pkg/config/config.go
Comment (hush-hush) [inline pkg/config/config.go]: No more removing trailing slash ?

**Review by hush-hush — CHANGES_REQUESTED** [review]

File: pkg/config/config.go
Comment (prognant) [reply pkg/config/config.go]: Logs_config.additional_endpoints uses a different structure than other addtional_endpoints (cf. https://github.com/DataDog/datadog-agent/blob/master/pkg/logs/config/endpoints.go#L13-L21),  and this loop was overwriting the config key logs_config.additional_endpoints with something inconsistent).
And it uses host & port fields so no need for trailing slash removal.

File: pkg/logs/config/config.go
Comment (prognant) [reply pkg/logs/config/config.go]: Will do.

**Review by olivielpeau — COMMENTED** [review]
LGTM, nice find about the (not yet shipped) regression 👍 

I'll let @hush-hush approve this once your last changes are done. Let's ask for a review from Logs-Intake as well for good measure

**Review by gaetan-deputier — APPROVED** [review]
lgtm!

**Review by hush-hush — APPROVED** [review]

**Pull request merged**
